src/drivers/imu/analog_devices/adis16507/ADIS16507.cpp

```diff
@@ -234,196 +285,154 @@ void ADIS16507::RunImpl()
 				ScheduleDelayed(SAMPLE_INTERVAL_US * 2);
 			}
 
-			bool success = false;
-
-			// TODO: review and test BURST_SEL = 1
-			//  16-Bit Burst Mode with BURST_SEL = 1
-			//  In 16-bit burst mode with BURST_SEL = 1, a burst contains
-			//  calibrated delta angle and delta velocity data in 16-bit format.
-
+			// struct __attribute__((packed)) BurstRead {
 			struct BurstRead {
 				uint16_t cmd;
-				uint16_t DIAG_STAT;
-				int16_t X_GYRO_OUT;
-				int16_t Y_GYRO_OUT;
-				int16_t Z_GYRO_OUT;
-				int16_t X_ACCL_OUT;
-				int16_t Y_ACCL_OUT;
-				int16_t Z_ACCL_OUT;
-				int16_t TEMP_OUT;
-				uint16_t DATA_CNTR;
+				uint16_t diag_stat;
+				int16_t x_gyro_out;
+				int16_t y_gyro_out;
+				int16_t z_gyro_out;
+				int16_t x_accl_out;
+				int16_t y_accl_out;
+				int16_t z_accl_out;
+				int16_t temp_out;
+				uint16_t data_cntr;
 				uint16_t checksum;
 			} buffer{};
 
 			// ADIS16507 burst report should be 176 bits
 			static_assert(sizeof(BurstRead) == (176 / 8), "ADIS16507 report not 176 bits");
 
-			buffer.cmd = static_cast<uint16_t>(Register::GLOB_CMD) << 8;
+			// Pg 20 of Datasheet
+			// 16-Bit Burst Mode with BURST_SEL = 0
+			buffer.cmd = 0x6800;
 			set_frequency(SPI_SPEED_BURST);
 
-			if (transferhword((uint16_t *)&buffer, (uint16_t *)&buffer, sizeof(buffer) / sizeof(uint16_t)) == PX4_OK) {
-
-				// Calculate checksum and compare
-
-				// Checksum = DIAG_STAT, Bits[15:8] + DIAG_STAT, Bits[7:0] +
-				//  X_GYRO_OUT, Bits[15:8] + X_GYRO_OUT, Bits[7:0] +
-				//  Y_GYRO_OUT, Bits[15:8] + Y_GYRO_OUT, Bits[7:0] +
-				//  Z_GYRO_OUT, Bits[15:8] + Z_GYRO_OUT, Bits[7:0] +
-				//  X_ACCL_OUT, Bits[15:8] + X_ACCL_OUT, Bits[7:0] +
-				//  Y_ACCL_OUT, Bits[15:8] + Y_ACCL_OUT, Bits[7:0] +
-				//  Z_ACCL_OUT, Bits[15:8] + Z_ACCL_OUT, Bits[7:0] +
-				//  TEMP_OUT, Bits[15:8] + TEMP_OUT, Bits[7:0] +
-				//  DATA_CNTR, Bits[15:8] + DATA_CNTR, Bits[7:0]
-				uint8_t *checksum_helper = (uint8_t *)&buffer.DIAG_STAT;
-
-				uint16_t checksum = 0;
-
-				for (int i = 0; i < 18; i++) {
-					checksum += checksum_helper[i];
-				}
+			if (transferhword((uint16_t *)&buffer, (uint16_t *)&buffer, sizeof(buffer) / sizeof(uint16_t)) != PX4_OK) {
+				perf_count(_bad_transfer_perf);
+				_failure_count++;
 
-				if (buffer.checksum != checksum) {
-					//PX4_DEBUG("adis_report.checksum: %X vs calculated: %X", buffer.checksum, checksum);
-					perf_count(_bad_transfer_perf);
-					perf_count(_perf_crc_bad);
+				if (_failure_count > 10) {
+					PX4_WARN("Consecutive failures!");
```

Comment:
```suggestion
					PX4_ERR("Consecutive failures!");
```

Although I'm not sure we should be printing either way.